### ISSUE TITLE
OAuthAccountNotLinked With only 1 user and 1 provider

### ISSUE THREAD

**Issue by rafiqumsieh0** [nextauthjs/next-auth#1446]
**Describe the bug**
I have only 1 user in the database and am using only 1 provider, which is Google. I am using MongoDB Atlas for the database. Initial signIn works fine and the user is created in the database, and a session is created. The bug occurs when I log that user out and I try to log in again. Now, when I try to sign the user in (User record is already in the database), it redirects me to the default sign in page: 

OAuthAccountNotLinked with the message : "To confirm your identity, sign in with the same account you used originally".

Every subsequent request will lead to the same result. 

If I delete the user from the database, it will work fine again.

I console logged the signIn callback, and Google is returning all the necessary data.

**Steps to reproduce**
Set up:

Safari browser
Google Provider
MongoDB Atlas Database

Steps:
1-) Try an initial log in with Google.
2-) After a successful user creation and session creation, log out.
3-) Try logging back in.

The bug should happen at step 3.

**Expected behavior**
I expected the signIn with Google to work fine since the initial SignIn works fine.

**Screenshots or error logs**
Just that default Sign In Page

**Additional context**
None at the moment.

**Feedback**
Thanks for the great library 

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [X] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by balazsorban44** [comment]
could you provide the error being thrown?

**Comment by rafiqumsieh0** [comment]
@balazsorban44 Thanks for the prompt response. The error is OAuthAccountNotLinked. I updated the comment to include more information including what is displayed on the default sign in page along with the error. Please let me know if you need any additional information.

**Comment by dr-starchildren** [comment]
I'm encountering the same error. I have a custom Signin page and am receiving the OAuthAccountNotLinked error.

**Comment by balazsorban44** [comment]
@rafiqumsieh0 you say you updated the original comment, but I can see no relevant change. I appreciate @dr-starchildren 's addition, but you reported the bug, so I would like to have a look at your configuration/reproduction as well.

@dr-starchildren which version are you using? I am noticing a `site` property in your options, which indicates a very old (2.x) version. Is this the case? Then I would start with upgrading to our latest release (as of this writing `3.10.1`) and you can safely remove the `site` property. It does nothing really. The important part is that the env variable is set.

**Comment by dr-starchildren** [comment]
@balazsorban44 I am using 3.10, just didn't realize I could remove the site. Thanks!

While my code allows for 3 providers, I've replicated with just a single provider. My problem is with Google only currently. I isolated Facebook and email verification and didn't have a problems signing in and out with them.

**Comment by dr-starchildren** [comment]
Ok, digging deeper...

I looked at where I was failing out in the debugger and saw that it stopped after getUserByProviderAccountId. In the debugger it shows the correct request going in 'Google' and the 'providerAccount.id'
However, I modified the code and saw that the result is undefined, even though it should match and account.

When I ran the same check with Facebook, I got a result. 

So...something in the getUserByProviderAccountId in next-auth/src/adapters/typeorm/index.js seems to be the problem

**Comment by dr-starchildren** [comment]
So it's failing in the  manager.findOne(User, { [idKey]: account.userId }) step of the get user account. It returns undefined, even though there is an account to be found.

```

async function getUserByProviderAccountId (providerId, providerAccountId) {
      debug('GET_USER_BY_PROVIDER_ACCOUNT_ID', providerId, providerAccountId)
      try {
        const account = await manager.findOne(Account, { providerId, providerAccountId })
        if (!account) { return null }
        return manager.findOne(User, { [idKey]: account.userId })
      } catch (error) {
        logger.error('GET_USER_BY_PROVIDER_ACCOUNT_ID_ERROR', error)
        return Promise.reject(new Error('GET_USER_BY_PROVIDER_ACCOUNT_ID_ERROR', error))
      }
    }
```

**Comment by dr-starchildren** [comment]
Maybe my code is out of sync...not sure how I just did a upgrade to 3.10, but my dist/adapters/models/account.js is different than the current main branch.

This is mine:
```
class Account {
  constructor(userId, providerId, providerType, providerAccountId, refreshToken, accessToken, accessTokenExpires) {
    this.compoundId = (0, _crypto.createHash)('sha256').update("".concat(providerId, ":").concat(providerAccountId)).digest('hex');
...
  } }
```

Current branch:
{
    // The compound ID ensures there is only one entry for a given provider and account
    ```
this.compoundId = createHash('sha256').update(`${providerId}:${providerAccountId}`).digest('hex')
   ...
  }
```

**Comment by dr-starchildren** [comment]
Okay, this was a long journey...

I missed that there was an accounts collection created and I was only resetting my users collection when doing my testing. So, a new account would be created successfully, however it would create a new ObjectID in Mongo under users. Then the findOne request in getUserbyProviderAccountId would return the first ObjectID, but that user no longer existed.

This function found the older ObjectID in accounts, which didn't match the new one in users.
`  const account = await manager.findOne(Account, { providerId, providerAccountId })`

This was entirely user error in resetting my database, but a possible solution is to return the latest ObjectID for a providerAccountId from the OAuth request, instead of the first. The risk is that poor database management would allow the Accounts collection to fill up with useless data.

**Comment by rafiqumsieh0** [comment]
Sorry guys I was afk. My code is very similar to dr-starchildren. Thanks to @dr-starchildren , I was able to solve the problem. I was not aware of the "accounts" collection that is used internally to keep track of users. I was deleting the "users" collection without clearing the "accounts' collection. I found multiple records in the "accounts" collection. That was probably created during initial setup and testing. Deleting both collections (users and accounts) solved my issue, and now I can log in with no issues so far. 

Thank you both @balazsorban44 @dr-starchildren
